### ISSUE TITLE
[BUG] `LogNormalDistributionLoss` behaves strange across different OS with `DecoderMLP`

### ISSUE THREAD

**Issue by phoeenniixx** [jdb78/pytorch-forecasting#1921]
**Describe the bug**

`LogNormalDistributionLoss` behaves strange across different OS with `DecoderMLP`. This specifically behaves strangely for `param-1` in `_decoder_mlp_pkg` used to test the model. For more info look at [this](https://github.com/sktime/pytorch-forecasting/pull/1908#issuecomment-3029226812) comment and the subsequent comments.

The loss-param fixture fails for different OS (like windows) but passes for OS like ubuntu. But this is also not always the case, when you make some change to `min_encoder_length` and `min_prediction_length`, it may pass for windows and fail for MacOS.

**To Reproduce**

Just remove the ` "tests:skip_by_name"` tag from the `DecoderMLP_pkg` and this will trigger the specific fixture that fails.

**Disclaimer**: If you test it locally on only one OS, it may pass as we've still not been able to diagnose the main issue and for which OS it fails for which param. This is quite random.

**Expected behavior**

The test should pass

**Additional context**

Look at [this](https://github.com/sktime/pytorch-forecasting/pull/1908#issuecomment-3029226812) comment and the subsequent comments.
and this discord thread: https://discord.com/channels/1075852648688930887/1080978709412069426/1391679541449068689

</details>

**Comment by phoeenniixx** [comment]
The specific param this test fails for is 
```python
dict(
                data_loader_kwargs=dict(min_prediction_length=2, min_encoder_length=2),
     ),
```

**Comment by harshsomankar123-tech** [comment]
Hi @phoeenniixx @fkiraly, 
I found and fixed an issue in LogNormalDistributionLoss where it raises a ValueError if the target tensor contains zeros. This happens because torch.distributions.LogNormal only accepts strictly positive values, but the base DistributionLoss.loss() method passes y_actual directly to log_prob() without validation. 

To resolve this, I overrode the loss() method and clamped y_actual to a minimum of 1e-4 before computing the loss, which matches the existing handling in prepare_test_inputs. This prevents crashes and instead returns a large but finite loss, which is expected during training. I verified the fix with a reproduction script and the integration test test_integration[DecoderMLP-base_params-0-LogNormalDistributionLoss], and everything passes now. 

If this approach looks good to you, I can go ahead and raise a PR for it. Let me know if you’d prefer the fix to be applied at the DistributionLoss level instead.
Thanks!

**Comment by phoeenniixx** [comment]
Thanks a lot @harshsomankar123-tech !
Please raise the PR and lets see how this turns out!
Don't forget to remove the tests from `tests:skip_by_name` tag!
